### ISSUE TITLE
Object Oriented Programming

### ISSUE THREAD

**Issue by pmaasz** [XRobots/openDog#22]
Hello James,
you write awesome functional code but have you ever considered to go further into oop? Maybe you discussed it in one of your episodes but I was just wondering why. Also you write a lot of different robot  related bits of software that share a few key features. You could also consider writing your own framework related to your projects to save some work. A benefit would be while improving code for a project you improve code for other robots simultanously.

**Comment by XRobots** [comment]
Yes I may do - there's not a lot that actually work well at the moment though - and a lot of dirty hacks!

**Comment by pmaasz** [comment]
yes but merging everything together can sometimes give a new perspective and problems deemed unsolvable suddenly become trivial. I don't want to push you towards something you don't want to do though :D.
